### PR TITLE
[Manure] Liquid manure storage bug fix and addition of missing separation efficiency

### PR DESCRIPTION
**Pull request by elle-andreen**
This PR addresses two small bugs in the manure module.

### Context
**Liquid storages:** As part of estimation methane emissions in slurry storage outdoor/underfloor and anaerobic lagoon, the ratio of bedding to manure non-degradable volatile solids (VS) in held manure is calculated and utilized to updated each VS fraction accordingly. If no manure is currently held in the storage, this calculation results in a divide by 0 error being returned, stopping the simulation. 

**Solid liquid separators:** The volatile solids removal efficiency (percent of entering volatile solids that are separated from the raw manure and recovered in the separated solids fraction) was not being applied correctly to bedding non-degradable VS in the liquid output. This resulted in the incorrect quantity of bedding VS leaving the separator with the liquid fraction.

### What
- Adds a check for 0 held bedding or manure non-degradable volatile solids and returns 0 in these cases to prevent a divide by 0 error when these types of storages are empty.
- Adding missing separation efficiency for bedding non-degradable volatile solids to separators.

### Why
Prevent an inappropriate divide by 0 error and correct the quantity of bedding VS in the liquid fraction leaving separators.

### How
See "What"

### Test plan
Manually confirmed separators were now calculating correct quantity of liquid fraction bedding ndVS. The sum of liquid and solid fraction bedding ndVS is equal to value of raw manure ndVS entering separator.

To test changes to liquid manure storages, I added a second storage (`slurry_storage_outdoor_lac2`) following the first storage (`slurry_storage_outdoor_lac1`) in the example freestall scenario. This configuration causes the second storage to be empty for 120 days, which previously triggered the divide by 0 bug. Those changes are reflected in the attached files that can be used to test this PR:

[example_freestall_processor_configs.json](https://github.com/user-attachments/files/24108183/example_freestall_processor_configs.json)
[example_freestall_processor_connections.json](https://github.com/user-attachments/files/24108179/example_freestall_processor_connections.json)

I tested the simulation the processor type of `slurry_storage_outdoor_lac2` set to slurry storage outdoor, slurry storage underfloor, and anaerobic lagoon to confirm all 3 processors are now working correctly. I also ran pytest and checked flake8 errors.

### Input Changes
No input changes.

### Output Changes
No output variable changes. Output variable value changes are limited to correcting the value of bedding ndVS in separator liquid fraction, which was previously overestimated.

#### Filter
```json
Manure.*
```

### CODE REVIEW

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py [hunk 1]
```diff
@@ -166,13 +166,23 @@ def _apply_methane_emissions(self, manure_temperature: float) -> tuple[float, fl
         total_methane = (
             storage_methane_from_degradable_volatile_solids + storage_methane_from_non_degradable_volatile_solids
         )
-        bedding_to_manure_non_degradable_volatile_solids_ratio = (
-            self._manure_to_process.bedding_non_degradable_volatile_solids
-            / (
-                self._manure_to_process.non_degradable_volatile_solids
-                + self._manure_to_process.bedding_non_degradable_volatile_solids
+
+        denominator = (
```
Comment (tomhuhh) [inline RUFAS/biophysical/manure/storage/anaerobic_lagoon.py]: Could we think of a more biological-oriented variable name here instead of 'denominator'?

**Review by tomhuhh — APPROVED** [review]
I believe these changes make sense. I have made some minor suggestions.

For the updates on the zero division error, I followed Elle's test plan and arrived at the same outcome. I didn't test with other edge case, but I'm open to suggestions for other edge cases to test on. 

Also, for the updates on the bedding non-degradable volatile solids calculations, I manually checked these variables below and the first two variables add up to the third variable, consistent with the manure processor configuration. 

`Manure.Separator.ScrewPress.screw_press_1.SeparatedSolids_manure_bedding_non_degradable_volatile_solids (kg)`
`Manure.Separator.ScrewPress.screw_press_1.SeparatedLiquid_manure_bedding_non_degradable_volatile_solids (kg)`
`Manure.SingleStreamHandler.ManualScraper.closeup_manual_scraper.manure_bedding_non_degradable_volatile_solids (kg)`

Screenshot 2025-12-11 at 2 38 06 PM

Thanks! Nice job, Elle!

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py [hunk 2]
```diff
@@ -166,13 +166,23 @@ def _apply_methane_emissions(self, manure_temperature: float) -> tuple[float, fl
...
+            self._manure_to_process.non_degradable_volatile_solids
+            + self._manure_to_process.bedding_non_degradable_volatile_solids
+        )
+
+        if (
+            self._manure_to_process.non_degradable_volatile_solids == 0
+            or self._manure_to_process.bedding_non_degradable_volatile_solids == 0
+            or denominator == 0
+        ):
+            bedding_to_manure_non_degradable_volatile_solids_ratio = 0
```
Comment (ew3361zh) [inline RUFAS/biophysical/manure/storage/anaerobic_lagoon.py]:
```suggestion
        if (
            self._manure_to_process.non_degradable_volatile_solids == 0.0
            or self._manure_to_process.bedding_non_degradable_volatile_solids == 0.0
            or denominator == 0.0
        ):
            bedding_to_manure_non_degradable_volatile_solids_ratio = 0.0
```

And doing this in the other storage child classes in the same spots will help prevent any additional MyPy errors.

**Review by ew3361zh — APPROVED** [review]
LGTM, one small recommendation on using floats instead of ints to prevent some mypy errors that should be applied the same across the three storages.
